testapp/ng2/app/async/async.component.ts

```diff
@@ -0,0 +1,72 @@
+import {bootstrap} from 'angular2/bootstrap';
+import {Component} from 'angular2/core';
+import {NgIf} from 'angular2/common';
+import {TimerWrapper} from 'angular2/src/facade/async';
```

Comment:
The TimeWrapper is used to allow for the timeout to be cancelable. After being canceled, if the timeout button is pressed again, it will wait for the entire timeout.